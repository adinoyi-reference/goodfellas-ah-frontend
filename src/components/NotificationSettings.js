import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updateNotification } from '../actions/notificationActions';

class NotificationSettings extends Component {
  state = {
    email: false,
    inApp: false,
  };

  handleToggleChange = async (e) => {
    const { updateNotification: toggleNotification } = this.props;
    let toggle = '';
    this.setState({ [e.target.name]: e.target.checked });
    if (e.target.checked) {
      toggle = 'on';
    } else {
      toggle = 'off';
    }
    await toggleNotification(e.target.name, toggle);
  };

  componentDidMount() {
    const { notification } = this.props;
    const currentNotification = notification.notification;
    const checkEmail = currentNotification.findIndex((setting) => setting === 'email');
    const checkinApp = currentNotification.findIndex((setting) => setting === 'inApp');
    if (checkEmail !== -1) {
      this.setState({
        email: true,
      });
    }
    if (checkinApp !== -1) {
      this.setState({
        inApp: true,
      });
    }
  }

  render() {
    const { email, inApp } = this.state;
    return (
      <div className="notification-settings-wrapper notification-panel">
        <div className="notification-settings-header">
          <h2>Notification Settings</h2>
        </div>
        <div className="notification-settings">
          <div className="notification-settings-email">
            <p>Email Notification</p>
            <label className="notification-switch">
              <input type="checkbox" onChange={this.handleToggleChange} name="email" checked={email} className="toggle-click" />
              <span className="slider round" />
            </label>
          </div>
          <div className="notification-settings-inApp">
            <p>In App Notification</p>
            <label className="notification-switch">
              <input type="checkbox" onChange={this.handleToggleChange} name="inApp" className="toggle-clickpu" checked={inApp} />
              <span className="slider round" />
            </label>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  notification: state.notification,
});

export default connect(
  mapStateToProps,
  { updateNotification },
)(NotificationSettings);
