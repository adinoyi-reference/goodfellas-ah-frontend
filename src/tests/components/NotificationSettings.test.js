import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Root from '../../root';
import NotificationSettings from '../../components/NotificationSettings';

let wrapped;


beforeEach(() => {
  wrapped = mount(
    <Root>
      <NotificationSettings />
    </Root>,
  );
});

describe('Notification UI', () => {
  describe('render features', () => {
    test('component should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
    it('should dropdown search bar filter section', () => {
      wrapped.find('.toggle-click').simulate('click');
      
    });
  });
});
