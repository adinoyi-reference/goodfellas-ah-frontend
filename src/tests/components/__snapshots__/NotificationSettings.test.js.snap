// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification UI render features component should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(NotificationSettings)>
      <NotificationSettings
        notification={
          Object {
            "articleError": "",
            "articleLoading": false,
            "articles": Array [],
            "authenticated": "",
            "error": Array [],
            "errorMessage": "",
            "notification": Array [],
            "notifications": Object {},
            "ownProfile": Object {},
            "profile": Object {
              "articles": Array [],
              "favorites": Array [],
              "followers": Array [],
              "following": Array [],
              "loading": true,
              "profile": Object {},
              "profileError": "",
              "profileView": "Following",
              "user": Object {},
            },
            "searchError": Array [],
            "searchResults": Array [],
            "singleArticle": null,
            "successMessage": "",
            "user": Object {},
          }
        }
        updateNotification={[Function]}
      >
        <div
          className="notification-settings-wrapper notification-panel"
        >
          <div
            className="notification-settings-header"
          >
            <h2>
              Notification Settings
            </h2>
          </div>
          <div
            className="notification-settings"
          >
            <div
              className="notification-settings-email"
            >
              <p>
                Email Notification
              </p>
              <label
                className="notification-switch"
              >
                <input
                  checked={false}
                  className="toggle-click"
                  name="email"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="slider round"
                />
              </label>
            </div>
            <div
              className="notification-settings-inApp"
            >
              <p>
                In App Notification
              </p>
              <label
                className="notification-switch"
              >
                <input
                  checked={false}
                  className="toggle-clickpu"
                  name="inApp"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="slider round"
                />
              </label>
            </div>
          </div>
        </div>
      </NotificationSettings>
    </Connect(NotificationSettings)>
  </Provider>
</_default>
`;
